src/manifest.c

```diff
@@ -1055,6 +1056,10 @@ int prune_manifest(struct manifest *manifest)
 			// LOG(file, "Skipping deleted boot file in manifest write", "component %s", manifest->component);
 			manifest->files = g_list_delete_link(manifest->files, list);
 			manifest->count--;
+		} else if (is_debuginfo(file->filename)) {
+			manifest->files = g_list_delete_link(manifest->files, list);
+			manifest->count--;
+			*pruned = true;
```

Comment:
This approach makes sense for the case when a file with `/usr/lib/debug` is already present in a manifest (deleted or not) and we want to remove it later on. However, with this change in place, files with `/usr/lib/debug` would never be added to a manifest in the first place...

So, I'm not sure this is going to behave well, in practice, because manifests will be written whenever a debuginfo file is encountered, regardless if any other "changes" were made to the bundle.